tests/main_test.go

```diff
@@ -0,0 +1,258 @@
+	if err != nil {
+		t.Fatalf("Cannot create stderr pipe: %v", err)
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		t.Fatalf("Error starting 'pebble run': %v", err)
+	}
+
+	done := make(chan struct{})
+	go func() {
+		defer close(logsCh)
+
+		readLogs := func(parser *servicelog.Parser) {
+			for parser.Next() {
+				if err := parser.Err(); err != nil {
+					t.Errorf("Cannot parse Pebble logs: %v", err)
+				}
+				select {
+				case logsCh <- parser.Entry():
+				case <-done:
+					return
+				}
+			}
+		}
+
+		// Both stderr and stdout are needed, because pebble logs to stderr
+		// while with "--verbose", services otuput to stdout.
+		stderrParser := servicelog.NewParser(stderrPipe, 4*1024)
+		stdoutParser := servicelog.NewParser(stdoutPipe, 4*1024)
+
+		// Channel to signal completion and close logsCh
+		done := make(chan struct{})
+		defer close(done)
+
+		go readLogs(stderrParser)
+		go readLogs(stdoutParser)
+
+		// Wait for both parsers to finish
+		<-done
+		<-done
+	}()
+
+	return logsCh
+}
+
+func waitForLogs(logsCh <-chan servicelog.Entry, expectedLogs []string, timeout time.Duration) error {
+	receivedLogs := make(map[string]struct{})
+
+	timeoutCh := time.After(timeout)
+	for {
+		select {
+		case log, ok := <-logsCh:
+			if !ok {
+				return errors.New("channel closed before all expected logs were received")
+			}
+
+			for _, expectedLog := range expectedLogs {
+				if _, ok := receivedLogs[expectedLog]; !ok && containsSubstring(log.Message, expectedLog) {
+					receivedLogs[expectedLog] = struct{}{}
+					break
+				}
+			}
+
+			allLogsReceived := true
+			for _, log := range expectedLogs {
+				if _, ok := receivedLogs[log]; !ok {
+					allLogsReceived = false
+					break
+				}
+			}
+
+			if allLogsReceived {
+				return nil
+			}
+
+		case <-timeoutCh:
+			missingLogs := []string{}
+			for _, log := range expectedLogs {
+				if _, ok := receivedLogs[log]; !ok {
+					missingLogs = append(missingLogs, log)
+				}
+			}
+			return errors.New("timed out waiting for log: " + strings.Join(missingLogs, ", "))
+		}
+	}
+}
+
+func containsSubstring(s, substr string) bool {
+	return strings.Contains(s, substr)
+}
+
+func waitForServices(t *testing.T, pebbleDir string, expectedServices []string, timeout time.Duration) {
+	t.Helper()
+
+	for _, service := range expectedServices {
+		waitForService(t, pebbleDir, service, timeout)
+	}
+}
+
+func waitForService(t *testing.T, pebbleDir string, service string, timeout time.Duration) {
+	t.Helper()
+
+	serviceFilePath := filepath.Join(pebbleDir, service)
+	timeoutCh := time.After(timeout)
+	ticker := time.NewTicker(time.Millisecond)
+	for {
+		select {
+		case <-timeoutCh:
+			t.Errorf("timeout waiting for service %s", service)
+			return
+
+		case <-ticker.C:
+			stat, err := os.Stat(serviceFilePath)
+			if err == nil && stat.Mode().IsRegular() {
+				os.Remove(serviceFilePath)
+				return
+			}
+		}
+	}
+}
+
+func isPortUsedByProcess(t *testing.T, port string, processName string) bool {
```

Comment:
This is another one that's very specific for a helper function. Per comments in the test, I think we should remove this helper and just use `http.Get` to do an actual HTTP request in the test.